sqlglot/generator.py

```diff
@@ -3616,6 +3618,22 @@ def dictrange_sql(self, expression: exp.DictRange) -> str:
     def dictsubproperty_sql(self, expression: exp.DictSubProperty) -> str:
         return f"{self.sql(expression, 'this')} {self.sql(expression, 'value')}"
 
+    def duplicatekeyproperty_sql(self, expression: exp.DuplicateKeyProperty) -> str:
+        expressions = self.expressions(expression, flat=True)
+        options = self.expressions(expression, key="options", flat=True, sep=" ")
+        options = f" {options}" if options else ""
+        return f"DUPLICATE KEY ({expressions}){options}"
+
+    # https://docs.starrocks.io/docs/sql-reference/sql-statements/data-definition/CREATE_TABLE/#distribution_desc
+    def distributedbyproperty_sql(self, expression: exp.DistributedByProperty) -> str:
+        expressions = self.expressions(expression, flat=True)
+        expressions = " (" + expressions + ")" if expressions else ""
```

Comment:
Nit: let's use `wrap` or an f-string, this way to wrap with parentheses is not very consistent with what we do.

Reply:
Got it